src/nosimg-enc.c

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+/*
+ * nosimg-enc.c - encode/decode "nos.img" image for XikeStor SKS8300 series
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <unistd.h>
+
+#define ENCODE_BLKLEN	0x100
+#define ENCODE_BLOCKS	2
+
+static uint32_t patterns[] = {
```

Comment:
Is this a collection of patterns of just a single pattern? This look like the encryption/decription key, so maybe `key`?

In any case it can be a `static const uint32_t[ENCODE_BLKLEN]`, since you only ever read from it and the size is fixed (and known).

Reply:
Indeed, I'll update to `key` and use `const` with explicit length.